pallets/grants/src/migrations.rs

```diff
@@ -0,0 +1,205 @@
+			} else {
+				log::info!(
+					"on_runtime_upgrade[{:#?}]=> Migration did not executed. This probably should be removed",
+					line!(),
+				);
+				T::DbWeight::get().reads(1)
+			}
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<(), &'static str> {
+			use frame_support::traits::OnRuntimeUpgradeHelpersExt;
+
+			let current = Pallet::<T>::current_storage_version();
+			let onchain = Pallet::<T>::on_chain_storage_version();
+
+			log::info!(
+				"pre_upgrade[{:#?}]=> with current storage version {:?} / onchain {:?}",
+				line!(),
+				current,
+				onchain
+			);
+
+			if current == 1 && onchain == 0 {
+				let pallet_prefix: &[u8] = b"Vesting";
+				let storage_item_prefix: &[u8] = b"VestingSchedules";
+
+				// Check number of entries, and set it aside in temp storage
+				let stored_data: Vec<_> =
+					storage_key_iter::<T::AccountId, Vec<VestingScheduleOf<T>>, Blake2_128Concat>(
+						pallet_prefix,
+						storage_item_prefix,
+					)
+					.collect();
+
+				let mapping_count = stored_data.len();
+				Self::set_temp_storage(mapping_count as u32, "mapping_count");
+
+				stored_data
+					.iter()
+					.for_each(|(_account, old_vesting)| assert!(old_vesting.len() <= T::MaxSchedule::get() as usize));
+
+				log::info!(
+					"pre_upgrade[{:#?}]=> VestingSchedules map count :: [{:#?}]",
+					line!(),
+					mapping_count,
+				);
+			} else {
+				log::info!(
+					"pre_upgrade[{:#?}]=> Migration did not executed. This probably should be removed",
+					line!(),
+				);
+			}
+			Ok(())
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade() -> Result<(), &'static str> {
+			use frame_support::traits::OnRuntimeUpgradeHelpersExt;
+
+			let current = Pallet::<T>::current_storage_version();
+			let onchain = Pallet::<T>::on_chain_storage_version();
+
+			log::info!(
+				"post_upgrade[{:#?}]=> with current storage version {:?} / onchain {:?}",
+				line!(),
+				current,
+				onchain
+			);
+
+			if current == 1 && onchain == 1 {
+				let pallet_prefix: &[u8] = b"Vesting";
+				let storage_item_prefix: &[u8] = b"VestingSchedules";
+
+				let stored_data: Vec<_> = storage_key_iter::<
+					T::AccountId,
+					BoundedVec<VestingScheduleOf<T>, T::MaxSchedule>,
+					Blake2_128Concat,
+				>(pallet_prefix, storage_item_prefix)
+				.collect();
+
+				let mapping_count = stored_data.len() as u32;
+
+				log::info!(
+					"post_upgrade[{:#?}]=> VestingSchedules map count :: [{:#?}]",
+					line!(),
+					mapping_count,
+				);
+
+				assert!(Some(mapping_count) == Self::get_temp_storage::<u32>("mapping_count"));
+
+				// // Check number of entries matches what was set aside in pre_upgrade
+				// if let Some(pre_mapping_count) = Self::get_temp_storage::<u32>("mapping_count") {
+				// 	assert!(pre_mapping_count == mapping_count);
+				// } else {
+				// 	log::info!(
+				// 		"post_upgrade[{:#?}]=> Pre-Migration did not executed. This probably should be removed",
+				// 		line!(),
+				// 	);
+				// }
```

Comment:
Do we need to keep the commented lines?

Reply:
no not required, statements are optimised.